src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py

```diff
@@ -545,6 +545,7 @@ def __call__(
         height: Optional[int] = None,
         width: Optional[int] = None,
         num_inference_steps: int = 50,
+        max_inference_steps: Optional[int] = None,
```

Comment:
```suggestion
        final_denoise_step: Optional[int] = None,
```

Maybe a bit better

Reply:
```python
    def timesteps_from_strength(
        self, strength: float, num_inference_steps: int
    ):
        """Retrieve values for `final_inference_step` and `begin_inference_step` from `strength`, `num_inference_steps`

        Args:
            strength (float): A traditional img2img strength between 0.0 and 1.0, with higher values resulting in greater
                                influence from the img2img model and lower values, more influence from the base model.
            num_inference_steps (int): The total number of inference steps to be taken.
        """
        # We need to invert the percentage. A strength of 0.0 should result in 100% of the inference steps.
        inverse_strength = 1.0 - strength
        final_inference_step = int(num_inference_steps * inverse_strength)
        begin_inference_step = final_inference_step
        return final_inference_step, begin_inference_step
```

here is a middle-ground i have struck, to allow for this rounding error to be handled in advance.